test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/controllers/MultipleAccountsCheckControllerSpec.scala

```diff
@@ -0,0 +1,118 @@
+        (mockAuthConnector
+          .authorise(
+            _: Predicate,
+            _: Retrieval[(Option[String] ~ Option[Credentials]) ~ Enrolments]
+          )(_: HeaderCarrier, _: ExecutionContext))
+          .expects(predicates, retrievals, *, *)
+          .returning(Future.successful(retrievalResponse()))
+        (mockEACDConnector
+          .getUsersWithPTEnrolment(_: String)(
+            _: ExecutionContext,
+            _: HeaderCarrier
+          ))
+          .expects(NINO, *, *)
+          .returning(createInboundResult(Some(UsersAssignedEnrolment1)))
+
+        val result = controller
+          .multipleAccountsCheck(testOnly.routes.TestOnlyController.successfulCall.url)
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
+
+        status(result) shouldBe SEE_OTHER
+        redirectLocation(result) shouldBe Some(
+          "/tax-enrolment-assignment-frontend/test-only/successful"
+        )
+      }
+    }
+
+    "user is signed in but PT enrolment is on another account" should {
+      "return OK" in {
+        (mockAuthConnector
+          .authorise(
+            _: Predicate,
+            _: Retrieval[(Option[String] ~ Option[Credentials]) ~ Enrolments]
+          )(_: HeaderCarrier, _: ExecutionContext))
+          .expects(predicates, retrievals, *, *)
+          .returning(Future.successful(retrievalResponse()))
+        (mockEACDConnector
+          .getUsersWithPTEnrolment(_: String)(
+            _: ExecutionContext,
+            _: HeaderCarrier
+          ))
+          .expects(NINO, *, *)
+          .returning(createInboundResult(Some(UsersAssignedEnrolment2)))
+
+        val result = controller
+          .multipleAccountsCheck(testOnly.routes.TestOnlyController.successfulCall.url)
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
+
+        status(result) shouldBe OK
+        contentAsString(result) shouldBe UCView()(fakeRequest, stubMessages(), appConfig).toString
+
+      }
+    }
+
+
+    "a no credentials exists in ES for a given enrolment key" should {
+      "return InternalServerError" in {
+        (mockAuthConnector
+          .authorise(
+            _: Predicate,
+            _: Retrieval[(Option[String] ~ Option[Credentials]) ~ Enrolments]
+          )(_: HeaderCarrier, _: ExecutionContext))
+          .expects(predicates, retrievals, *, *)
+          .returning(Future.successful(retrievalResponse()))
+        (mockEACDConnector
+          .getUsersWithPTEnrolment(_: String)(
+            _: ExecutionContext,
+            _: HeaderCarrier
+          ))
+          .expects(NINO, *, *)
+          .returning(createInboundResultError(UnexpectedResponseFromEACD))
+
+        val result = controller
+          .multipleAccountsCheck(testOnly.routes.TestOnlyController.successfulCall.url)
+          .apply(buildFakeRequestWithSessionId("GET", "Not Used"))
+
+        status(result) shouldBe INTERNAL_SERVER_ERROR
+      }
+    }
+  }
```

Comment:
This is not an internal error case! This case is just a user with multiple credential where non have a PT enrolment.